ui/app/src/nh-config/dimension-list.ts

```diff
@@ -0,0 +1,265 @@
+    } catch (error) {
+      console.log('Error fetching dimension details: ', error);
+    }
+  }
+
+  async fetchRangeEntriesFromHashes(rangeEhs: EntryHash[]) {
+    const response = await Promise.all(rangeEhs.map(eH => this.fetchRange(eH)))
+    this._rangeEntries = response.map((payload, index) => {
+      try {
+        return { ...decode(payload.entry.Present.entry) as Range, range_eh: rangeEhs[index]}
+      } catch (error) {
+        console.log('Error decoding range payload: ', error);
+      }
+    }) as Array<Range & { range_eh: EntryHash }>;
+  }
+
+  async firstUpdated(_changedProperties: PropertyValueMap<any> | Map<PropertyKey, unknown>) {
+    await this.fetchDimensionEntries()
+    await this.fetchRangeEntries()
+    // NOTE: The following retrieves a mapping for method entry hashes to their respective input/output dimension hashes, but ONLY for those registered in an appletConfig.
+    if(typeof this._methodMapping.value == 'undefined') return;
+    
+    // If there is a method mapping (from applet config) it will be included here
+    const dimensions : any = [];
+    for (const [methodEh,{inputDimensionEh}] of Object.entries(this._methodMapping.value)) {
+      const response = await this.fetchDimension(inputDimensionEh);
+      try {
+        const dimension = decode(response.entry.Present.entry) as any;
+        dimension.methodEh = methodEh;
+        dimensions.push(dimension)
+      } catch (error) {
+        console.log('Error decoding dimension payload: ', error);
+      }
+    }
+    this.methodInputDimensions = [...this.methodInputDimensions, ...this._dimensionEntries, ...dimensions];
+
+    // Select the first item in the list:
+    this.dispatchEvent(new CustomEvent("input-dimension-selected", {
+      detail: { range: this._rangeEntries[this._rangeEntries.length-1], dimensionEh: this._dimensionEntries[this._dimensionEntries.length-1].dimension_eh },
+      bubbles: true,
+      composed: true,
+    }));
+    this.dimensionSelected = true;
+  }
+
+  protected updated(_changedProperties: PropertyValueMap<any> | Map<PropertyKey, unknown>): void {
+    if(_changedProperties.has('methodInputDimensions') && !!_changedProperties.get('methodInputDimensions')) {
+      this.filteredMethodInputDimensions = this.methodInputDimensions.filter(dimension => dimension?.methodEh);
+    }
+  }
+
+  renderRangeDetails(range: Range & { range_eh: EntryHash } | undefined) : TemplateResult {
+    if(typeof range == "undefined") return html``;
+    return html`
+      <nh-card .theme=${"light"} .title=${range.name} >
+        <h3><em>Kind:</em></h3>
+        <div>${Object.keys(range.kind)[0]}</div>
+        <h3><em>Min:</em></h3>
+        <span>${Object.values(range.kind)[0].min}</span>
+        <h3><em>Max:</em></h3>
+        <span>${Object.values(range.kind)[0].max}</span>
+      </nh-card>
+    `
+  }
+
+  resetSelectedInputDimensionIndex() {
+    this._selectedInputDimensionIndex = 0;
+  }
+
+  render() {
```

Comment:
Please break this up into separate functions for displaying things. I would suggest only using trinary expressions when outputting possibly `undefined` or `null`. If statements with functions are easier to read that inline trinaries in html literals.